### ISSUE TITLE
bring distinction between main tools cursors

### ISSUE THREAD

**Issue by gabuzom** [mapeditor/tiled#3192]
Tiled is a wonderful piece of free/libre software! thanks immensely for it!!

Yet there is one minor frustration i experience while using it, and juggling between the main tools using keyboard shortcuts: brush/eraser/rectangular selection... sometimes i am so absorbed that i don't realize, after switching to another operation, which was selected (and therfeore erase instead of brushing, or brush instead of...). Sure it's as simple as looking at the toolbar, but when you're really absorbed in your design you never do...

How about: 
* brush has this blue box-like cursor+selection
* eraser has the same but a *red* box-like cursor+selection
* rectangular has the same but a *green* box-like cursor+selection?

this way one would instanteously be able to shift their mind from one tool to the other 

of course if i am the only one thinking it would be a good idea it probably means it is not worth implementing ;)

cheers!

**Comment by bjorn** [comment]
> of course if i am the only one thinking it would be a good idea it probably means it is not worth implementing ;)

No, this is a known usability problem so thank you for opening an issue about it! I'm just not sure whether distinguishing the tools just by the color of their highlight is the right approach. Using a red highlight for eraser seems sensible, but using green for the shape tool feels arbitrary. Currently, the highlight follows the configurable (if not using system style) selection color, so the question is also whether we should entirely drop that.

Personally, I've been more thinking about introducing different mouse cursors, which is what seems more common among graphical software. Of course creating mouse cursors is more work than setting up some different colors, but maybe we could use some from Inkscape for example.
